src/main/include/util/TrajectoryManager.hpp

```diff
@@ -19,11 +19,11 @@ class TrajectoryManager {
   const Trajectory& GetTrajectory(const std::string& name) const;
 
  private:
-  void LoadTrajectories();
+  std::map<std::string, Trajectory> LoadTrajectories();
 
   static Trajectory LoadFile(const std::filesystem::path& trajPath);
 
-  std::map<std::string, Trajectory> m_trajectories;
+  const std::map<std::string, Trajectory> m_trajectories;
```

Comment:
This const is redundant. The only public function available is GetTrajectory(), which returns a const ref already.

Reply:
So the main purpose of making PR in the first place doesn't make sense? I remember before you said taking the const reference of a regular `std::map` was unsafe. I changed it to `const std::map` to avoid that problem. It seems like this conflicts with the advice you gave before.

Reply:
If you write the implementation correctly, it should be fine. I get that. But why risk it?

Reply:
> I remember before you said taking the const reference of a regular std::map was unsafe.

I don't remember saying that, but if I had to guess it had to do with accessing global objects directly between translation units. Returning a reference to something inside the global object has no effect on when the global object's constructor is run. You clarified it wasn't related to the global variables in the constants header.

> If you write the implementation correctly, it should be fine. I get that. But why risk it?

You're not risking anything. It's clear from the interface you already have that the interaction you're worried about (the user modifying the trajectory map) is impossible.

Reply:
From FRC Discord:

> you're getting a pointer to a trajectory, but that can get invalidated if you add more trajectories to the std::map it came from since the container could reallocate (but won't necessarily do so every time)

Reply:
The contrapositive of that statement is if you don't add to the container, references won't be invalidated so you can use them. The API you've implemented prevents additions after startup, so references obtained after that are safe.

Reply:
Ok so you were more worried about that being exposed in the API, but it's not like that, so it's ok?

Reply:
Yea.